### ISSUE TITLE
[BUG REPORT]

### ISSUE THREAD

**Issue by randomairborne** [mcpyproject/McPy#19]
**Describe the bug**
System not working 

**To Reproduce**
root | INFO | Trying to initialize the Blackfire probe
root | INFO | Blackfire not installed: passing
root | INFO | Starting queues...
root | INFO | Started queues!
root | INFO | Trying to find number of available cores
root | WARNING | Falling back to multiprocessing cpu_count to calc cores. Most likely getaffinity is not supported on your OS
root | INFO | Found 2 cores available!
root | INFO | Starting worker ID 24407
root | INFO | Started worker.
root | INFO | Starting networking worker
Traceback (most recent call last):
  File "/Users/Grant/McPy/main.py", line 238, in <module>
    main()
  File "/Users/Grant/McPy/main.py", line 221, in main
    networkingProcess.start()
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/multiprocessing/process.py", line 121, in start
    self._popen = self._Popen(self)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/multiprocessing/context.py", line 224, in _Popen
    return _default_context.get_context().Process._Popen(process_obj)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/multiprocessing/context.py", line 284, in _Popen
    return Popen(process_obj)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/multiprocessing/popen_spawn_posix.py", line 32, in __init__
    super().__init__(process_obj)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/multiprocessing/popen_fork.py", line 19, in __init__
    self._launch(process_obj)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/multiprocessing/popen_spawn_posix.py", line 47, in _launch
    reduction.dump(process_obj, fp)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/multiprocessing/reduction.py", line 60, in dump
    ForkingPickler(file, protocol).dump(obj)
TypeError: cannot pickle '_cffi_backend.FFI' object
root | INFO | Trying to initialize the Blackfire probe
root | INFO | Blackfire not installed: passing
root | INFO | Starting queues...
root | INFO | Started queues!
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/multiprocessing/spawn.py", line 116, in spawn_main
    exitcode = _main(fd, parent_sentinel)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/multiprocessing/spawn.py", line 126, in _main
    self = reduction.pickle.load(from_parent)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/multiprocessing/synchronize.py", line 110, in __setstate__
    self._semlock = _multiprocessing.SemLock._rebuild(*state)
FileNotFoundError: [Errno 2] No such file or directory

**Expected behavior**

**Screenshots**

**Host system:**
 - OS: [Macos Catalina Latest]
 - OS version: [10.15.6]
 - Python version [3.8.5/Pycharm 2020.2] 

**Additional context**

**Comment by tazz4843** [comment]
MacOS is known not to have a working shared semaphore implementation, so McPy will not work on MacOS systems. I will try to catch as many of these errors as possible to warn users of this issue.

**Comment by ntoskrnl4** [comment]
```
Traceback (most recent call last):
    File "/Users/Grant/McPy/main.py", line 238, in 
        main()
    File "/Users/Grant/McPy/main.py", line 221, in main
        networkingProcess.start()
[cut]
    File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/multiprocessing/popen_spawn_posix.py", line 47, in _launch
        reduction.dump(process_obj, fp)
    File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/multiprocessing/reduction.py", line 60, in dump
        ForkingPickler(file, protocol).dump(obj)
TypeError: cannot pickle '_cffi_backend.FFI' object
```

This exception is the exact same exception and traceback I get when trying to start the server on my machine (Windows). It seems whatever mechanism is used for the threads is Linux-only...? Perhaps Twisted or Quarry don't properly support it or don't properly support 3.8 or something.

**Comment by ntoskrnl4** [comment]
I am fairly sure we are doing something wrong here that's making this not working. It should be possible to do IPC by the way of event queues, regardless of the platform.

I'll do some digging tomorrow and see what I can find about all of this

**Comment by kevdagoat** [comment]
Hey @ntoskrnl4 

I do not have that issue running on Win10 1903

**Comment by tazz4843** [comment]
I just realized what a `_cffi_backend.FFI` object is exactly: some *nix systems end up having issues with `twisted`, since it imports `cryptography`, which imports another package from somewhere on the system: it can't pickle a entire package, so it throws that error.

**Comment by 0ddlyoko** [comment]
Fixed in #20
This was because an object (https://github.com/tazz4843/McPy/blob/282a66967cbaf1360413835d7bb92d8d0937180b/main.py#L264) was created in another Thread as the network logic.
